File: pkg/domain/model/license.go
package model

import "time"

type LicensePageReq struct {
	Page         int    `json:"page" query:"page" form:"page" validate:"required,min=1"`
	Size         int    `json:"page_size" query:"page_size" form:"page_size" validate:"required,min=1,max=100"`
	Domain       string `json:"domain" query:"domain" form:"domain"`
	CustomerName string `json:"customer_name" query:"customer_name" form:"customer_name"`
	Status       *int   `json:"status" query:"status" form:"status"`
}

type LicenseCreateReq struct {
	Domain       string    `json:"domain" validate:"required"`
	CustomerName string    `json:"customer_name"`
	ExpireDate   time.Time `json:"expire_date" validate:"required"`
}

type LicenseUpdateReq struct {
	Domain       string    `json:"domain,omitempty"`
	LicenseKey   string    `json:"license_key,omitempty"`
	CustomerName string    `json:"customer_name,omitempty"`
	ExpireDate   time.Time `json:"expire_date,omitempty"`
	Status       int       `json:"status,omitempty"`
}

type LicenseResp struct {
	ID           int       `json:"id"`
	CreatedAt    LocalTime `json:"created_at"`
	UpdatedAt    LocalTime `json:"updated_at"`
	Domain       string    `json:"domain"`
	LicenseKey   string    `json:"license_key"`
	CustomerName string    `json:"customer_name"`
	ExpireDate   LocalTime `json:"expire_date"`
	Status       int       `json:"status"`
}

type LicenseVerifyReq struct {
	Domain string `json:"domain" validate:"required"`
}

type LicenseVerifyResp struct {
	Valid        bool   `json:"valid"`
	CustomerName string `json:"customer_name"`
	ExpireDate   string `json:"expire_date"`
	Message      string `json:"message"`
}

File: ui/src/api/infra/license.ts
import {http} from '@/utils/http'
import { BASE_URL, type ApiResponse, type TableResponse } from '@/api/utils'

export interface License {
  id: number
  created_at: string
  updated_at: string
  domain: string
  license_key: string
  customer_name: string
  expire_date: string
  status: number
}

export interface LicensePageParams {
  page: number
  page_size: number
}

export interface LicenseCreateReq {
  domain: string
  customer_name: string
  expire_date: string
}

export interface LicenseUpdateReq {
  domain?: string
  license_key?: string
  customer_name?: string
  expire_date?: string
  status?: number
}

export interface LicenseVerifyReq {
  domain: string
}

export interface LicenseVerifyResp {
  valid: boolean
  customer_name: string
  expire_date: string
  message: string
}

export const getLicensePage = (params: LicensePageParams) => {
  return http.request<TableResponse<License>>('get', `${BASE_URL}/v1/license/page`, { params })
}

export const queryLicense = (id: number) => {
  return http.request<ApiResponse<License>>('get', `${BASE_URL}/v1/license/query/${id}`)
}

export const createLicense = (data: LicenseCreateReq) => {
  return http.request<ApiResponse<License>>('post', `${BASE_URL}/v1/license/create`, { data })
}

export const updateLicense = (id: number, data: LicenseUpdateReq) => {
  return http.request<ApiResponse<License>>('put', `${BASE_URL}/v1/license/update/${id}`, { data })
}

export const deleteLicense = (id: number) => {
  return http.request<ApiResponse<null>>('delete', `${BASE_URL}/v1/license/delete/${id}`)
}

export const verifyLicense = (data: LicenseVerifyReq) => {
  return http.request<ApiResponse<LicenseVerifyResp>>('post', `${BASE_URL}/v1/license/verify`, { data })
}
